app/core/blockchain/state_updater.py

```diff
@@ -33,99 +33,73 @@
 def update_db_states(cur, block):
     newblockindex = block['index'] if 'index' in block else block['block_index']
     transactions = block['text']['transactions']
-    # last_block_cursor = cur.execute(
-    #     f'''SELECT block_index FROM blocks ORDER BY block_index DESC LIMIT 1''')
-    # last_block = last_block_cursor.fetchone()
-    # if newblockindex != last_block[0]:
-    #     print("The latest block index does not match given previous index")
-    #     return False
-    #    latest_index = cur.execute('SELECT MAX(block_index) FROM blocks')
+
     add_tx_to_block(cur, newblockindex, transactions)
 
     if 'creator_wallet' in block and block['creator_wallet'] is not None:
         add_block_reward(cur, block['creator_wallet'], newblockindex)
 
-    collated_txns = simplify_transactions(cur, transactions)
     global simplified_transactions
```

Comment:
@Prannav-k can we avoid globals? This is very unsafe and cause to inconsistencies if two threads ever run at the same time. Now that we're refactoring, best to change.